src/xtgeo/grid3d/grid_property.py

```diff
@@ -57,7 +76,7 @@
 # ======================================================================================
 
 
-def _data_reader_factory(fformat):
+def _data_reader_factory(fformat: str) -> Callable:
```

Comment:
Would like a Literal here, but seems like something that should  live in a xtgeo.types file.

Reply:
Agreed, fortunately `fformat` as a parameter name is consistent throughout the codebase. I'll remember highlight this specifically in an issue so it can be redressed in the future